src/main/java/net/glowstone/entity/GlowPlayer.java

```diff
@@ -901,39 +904,37 @@ protected void jump() {
      * Process and send pending BlockChangeMessages.
      */
     private void processBlockChanges() {
-        synchronized (blockChanges) {
-            List<BlockChangeMessage> messages = new ArrayList<>(blockChanges);
-            blockChanges.clear();
-            // separate messages by chunk
-            // inner map is used to only send one entry for same coordinates
-            Map<Key, Map<BlockVector, BlockChangeMessage>> chunks = new HashMap<>();
-            for (BlockChangeMessage message : messages) {
-                if (message != null) {
-                    Key key = GlowChunk.Key.of(message.getX() >> 4, message.getZ() >> 4);
-                    if (canSeeChunk(key)) {
-                        Map<BlockVector, BlockChangeMessage> map = chunks
-                                .computeIfAbsent(key, k -> new HashMap<>());
-                        map.put(new BlockVector(message.getX(), message.getY(), message
-                                .getZ()), message);
-                    }
-                }
+        // separate messages by chunk
+        // inner map is used to only send one entry for same coordinates
+        Map<Key, Map<BlockVector, BlockChangeMessage>> chunks = new HashMap<>();
+        while (true) {
```

Comment:
`while (!blockChanges.isEmpty())` ?

Removes the need for the null check afterwards.